### PR TITLE
add setup documentation to the README

### PR DESCRIPTION
**Pull request by seport**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,29 @@
...
+  brew install mongodb
+  ```
+- install packages
+  ```
+  npm install
+  ```
+- install sass
+  ```
+  sudo gem install sass
+  ```
```
Comment (Reltre) [inline README.md]:
```suggestion

### Build & Setup App

Generate assets with webpack
```
npm run build
```
Start the database service
```
npm run restart-db
```
```

**Comment by Reltre** [comment]
@seport I've updated the README and added an updated mongo config and shell script. Things should work well with the current setup. Let me know what you think.

File: README.md [hunk 2]
```diff
@@ -1,14 +1,50 @@
...
 
-## Install dependencies
-- `brew install node`
-- `npm install`
-- `sudo gem install sass`
+You only have to do this once.
+
+- ask in slack channel for `secrets.js` file
+- add `conf/secrets.js` file
+- add `.env`
```
Comment (jgreub) [inline README.md]:
```suggestion
- ask in the slack channel for the `secrets.js` file
- add the `secrets.js` file to the `conf/` directory
- ask in the slack channel for the `.env` file
- add the `.env` file to the project root directory
```

File: README.md [hunk 3]
```diff
@@ -1,14 +1,50 @@
...
-- add a file named env.js in conf folder, with contents of `module.exports = 'dev';`
 
-## Install dependencies
-- `brew install node`
-- `npm install`
-- `sudo gem install sass`
+You only have to do this once.
+
+- ask in slack channel for `secrets.js` file
+- add `conf/secrets.js` file
```
Comment (jgreub) [inline README.md]:
```suggestion
- add the `secrets.js` file to the `conf/` directory
```

File: README.md [hunk 4]
```diff
@@ -1,14 +1,50 @@
...
-- ask wcrest for secrets.js file, and put into conf folder
-- add a file named env.js in conf folder, with contents of `module.exports = 'dev';`
 
-## Install dependencies
-- `brew install node`
-- `npm install`
-- `sudo gem install sass`
+You only have to do this once.
+
+- ask in slack channel for `secrets.js` file
```
Comment (jgreub) [inline README.md]:
```suggestion
- ask in the slack channel for the `secrets.js` file
```

File: scripts/provision_mongo.sh [hunk 5]
```diff
@@ -0,0 +1,25 @@
+# expects a firewall rule opening port 27018 to application servers
+source .env
+
+# Setup database directory
+sudo mkdir -p /data/db
+sudo chmod -R 757 /data/db
+
+# Stop an existing mongo instance
+pgrep mongod | xargs kill -2
```
Comment (jgreub) [inline scripts/provision_mongo.sh]:
```suggestion
pgrep mongod | xargs kill -2
sleep 1
```

File: scripts/provision_mongo.sh [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+# Install TLS certificate, configure mongo to use it
+cd /etc/ssl/
+sudo openssl req -newkey rsa:2048 -new -x509 -days 365 -nodes -out mongodb-cert.crt -keyout mongodb-cert.key -subj "/C=US/ST=California/L=San Francisco/O=helloGov/OU=Engineering/CN=data.hellogov.org"
+sudo sh -c "cat mongodb-cert.key mongodb-cert.crt > mongodb.pem"
+sudo scp mongodb.pem /tmp/mongodb.pem
+sudo cp -r "$LOCAL_DIR/conf/mongod.conf" /etc
+
+# Test secure connections by creating application user
+pgrep mongod | xargs kill -2
```
Comment (jgreub) [inline scripts/provision_mongo.sh]:
```suggestion
# Test secure connection and create standard db user
```

File: scripts/provision_mongo.sh [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+mongo localhost:27017/admin --eval "db.createUser({user:'$DB_ADMIN_USER',pwd:'$DB_ADMIN_PASSWORD',roles:[{role:'userAdminAnyDatabase',db:'admin'}, 'readWriteAnyDatabase']})"
+
+# Install TLS certificate, configure mongo to use it
+cd /etc/ssl/
+sudo openssl req -newkey rsa:2048 -new -x509 -days 365 -nodes -out mongodb-cert.crt -keyout mongodb-cert.key -subj "/C=US/ST=California/L=San Francisco/O=helloGov/OU=Engineering/CN=data.hellogov.org"
+sudo sh -c "cat mongodb-cert.key mongodb-cert.crt > mongodb.pem"
+sudo scp mongodb.pem /tmp/mongodb.pem
+sudo cp -r "$LOCAL_DIR/conf/mongod.conf" /etc
+
+# Test secure connections by creating application user
```
Comment (jgreub) [inline scripts/provision_mongo.sh]:
```suggestion
# Stop an existing mongo instance
pgrep mongod | xargs kill -2
sleep 1
```

File: scripts/provision_mongo.sh [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+source .env
+
+# Setup database directory
+sudo mkdir -p /data/db
+sudo chmod -R 757 /data/db
+
+# Stop an existing mongo instance
+pgrep mongod | xargs kill -2
+
+# Create admin user
```
Comment (jgreub) [inline scripts/provision_mongo.sh]: # Create admin db user

**Comment by jgreub** [comment]
I also had to update the `index.js` file with these two changes:

1) line 38 `mongoose.connect(mongoUri, { useMongoClient: true });`
2) line 43 `mongoose.connection.once('openUri', function callback() {`

File: README.md [hunk 9]
```diff
@@ -1,14 +1,51 @@
...
-- `sudo gem install sass`
+You only have to do this once.
+
+- ask in the slack channel for the `secrets.js` file
+- add the `secrets.js` file to the `conf/` directory
+- ask in the slack channel for the `.env` file
+- add the `.env` file to the project root directory
+  ```
+  $ touch conf/secrets.js
+  ```
```
Comment (jgreub) [inline README.md]:
```suggestion

```

File: README.md [hunk 10]
```diff
@@ -1,14 +1,51 @@
...
-- `npm install`
-- `sudo gem install sass`
+You only have to do this once.
+
+- ask in the slack channel for the `secrets.js` file
+- add the `secrets.js` file to the `conf/` directory
+- ask in the slack channel for the `.env` file
+- add the `.env` file to the project root directory
+  ```
+  $ touch conf/secrets.js
```
Comment (jgreub) [inline README.md]:
```suggestion

```

File: README.md [hunk 11]
```diff
@@ -1,14 +1,51 @@
...
-- `brew install node`
-- `npm install`
-- `sudo gem install sass`
+You only have to do this once.
+
+- ask in the slack channel for the `secrets.js` file
+- add the `secrets.js` file to the `conf/` directory
+- ask in the slack channel for the `.env` file
+- add the `.env` file to the project root directory
+  ```
```
Comment (jgreub) [inline README.md]:
```suggestion

```

**Pull request merged**